### PR TITLE
fix: group help buttons

### LINKED ISSUE #37627: group lesson buttons.
To have a cleaner UI and to encourage users to attempt challenges before asking for help, we need to group the help buttons in lessons.

There should only be three buttons in each lesson: "run the tests", "reset all code", "get help", and their wrapper element should get a ```button-group``` class to space the buttons out and improve lighthouse scores. 

"get hint", "ask for help", "watch a video" buttons should be grouped under the "get help" button and displayed as a dropdown similar to the certification section on settings, or it could be a "dropup" to account for lessons that have a couple of tests or for mobile views. 

Screen Shot 2019-10-31 at 11 55 39 AM

No assignment needed. please proceed with submitting your solution in a pr.

### PR DESCRIPTION
**Pull request by DerrykBoyd**
group help buttons on challenges and add styling for mobile view.

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes #37627

### CODE REVIEW

**Comment by ahmadabdolsaheb** [comment]
Thank you for looking into this optimization.

I pulled your pr locally and it looks good for the most part. however, the instruction panel scrolls pass the content, the test items are missing and the scrimba video does not show up in the drop up menu.

image


Could you please fix the above-mentioned issues so we could proceed with reviewing your pr?

**Comment by dboydgit** [comment]
@ahmadabdolsaheb Thanks for the review. I think I've fixed the first two issues. I don't see the issue with the video on my end. Are you sure the challenge in your screenshot has a video?

File: client/src/templates/Challenges/components/tool-panel.css [hunk 1]
```diff
@@ -14,8 +14,35 @@
...
+
+.tool-panel-group-mobile .btn-group {
+  display: block;
+  width: 100%;
+  margin: 0 2px 0 2px;
+}
+
+.tool-panel-group-mobile .btn {
+  margin-bottom: 0 !important;
+}
```
Comment (ahmadabdolsaheb) [inline client/src/templates/Challenges/components/tool-panel.css]: do we need this, it seems like line 18 is taking care of this?

Comment (ahmadabdolsaheb) [reply client/src/templates/Challenges/components/tool-panel.css]: other than that it LGTM

Comment (dboydgit) [reply client/src/templates/Challenges/components/tool-panel.css]: I think there was some styling in global.css taking precedence that caused the button spacing to be off.

Comment (ahmadabdolsaheb) [reply client/src/templates/Challenges/components/tool-panel.css]: I tried it locally and gitpod. removing it did not change the style. could you reconfirm and remove if not necessary? thank you

Comment (dboydgit) [reply client/src/templates/Challenges/components/tool-panel.css]: This is the issue I was seeing. When removed there is a 10px margin below the buttons, leading to page overflow and the buttons can be scrolled past.

Screenshot from 2019-11-08 12-57-31

Actually even without the bottom margin, there is still a 1-2 px overflow, not really noticeable if on a mobile device, but can lead to a double vertical scrollbar on desktop. I can't figure out what's causing it though.

Comment (ahmadabdolsaheb) [reply client/src/templates/Challenges/components/tool-panel.css]: hmm, for some reason i could not reproduce either.

**Review by ahmaxed — APPROVED** [review]
LGMT, would appreciate it if the second reviewer could double check if 
```
.tool-panel-group-mobile .btn {
  margin-bottom: 0 !important;
}
```
is necessary.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Some observations:

On mobile it looks a little cramped.  There's quite a lot of screen available and we do want to make it easy to tap on stuff.  Also, the button's styling doesn't toggle back when you tap a second time (only when you select something else):

MenuMobile

On desktop, you can see the other two buttons slightly at the sides.  I think it would look better if they were covered entirely.

image

Finally, shouldn't they look like buttons - no underlining and perhaps borders separating them?  I know they are links, but it the underlining feels out of place since we generally don't use it.  I feel like they would look better if they used the original style, but only appeared when you expand things.

@ahmadabdolsaheb could you weigh in on whether or not you think these are reasonable suggestions?

**Comment by ahmadabdolsaheb** [comment]
@ojeytonwilliams thanks for looking into this. It seems like the convention for the button group drop downs is displaying them in a list like column. 

while I admit hey are not the most aesthetic buttons, having the underline helps with displaying them as active text and is somehow consistent with the rest of the UI. 

I completely agree with you regarding mobile. maybe we could say 'help' instead of 'get help' for mobile and go wider with the group so it could display the full text of the buttons?

buttons group drop up could also be wider to cover the other buttons.

**Comment by ojeytonwilliams** [comment]
@ahmadabdolsaheb no problem at all.  @dboydgit if you've got a moment, could you implement Ahmad's suggestions?

**Comment by dboydgit** [comment]
Just got back from vacation 😃 Should be able to make those updates this weekend.

**Comment by dboydgit** [comment]
Latest commit should address all suggestions to date except for the button hover colour on mobile. Mobile devices will maintain the `:hover`state until there is another click outside the button.

File: client/src/templates/Challenges/components/tool-panel.css [hunk 2]
```diff
@@ -11,11 +11,43 @@
...
+  width: calc(100% + 2px);
+  text-align: center;
+  min-width: 240px;
+  left: -1px;
+}
+
+.tool-panel-group .dropdown-menu a {
+  padding: 0.5rem;
+}
+
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/components/tool-panel.css]:
```suggestion

/* Prevent 'sticky hover' effects for mobile devices */
@media (hover: none) {
  .dropdown-toggle.btn-primary:hover {
    color: var(--secondary-color);
    background-color: var(--quaternary-background);
  }
}

```
This should get rid of the hover effect on mobile.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
@dboydgit it looks loads better now, nice job!

Could you take a look at my suggested fix for the hover effect and commit it if you think it's the right solution?

File: client/src/templates/Challenges/components/tool-panel.css (hunk 2 above)
Comment (dboydgit) [reply client/src/templates/Challenges/components/tool-panel.css]: That works great, thanks!

**Review by ojeytonwilliams — APPROVED** [review]
LGTM!

**Comment by raisedadead** [comment]
@ahmadabdolsaheb please merge when you have done the QA on this

**Comment by ahmadabdolsaheb** [comment]
@dboydgit Thanks a lot for working on this feature, and @ojeytonwilliams for the review. 🎆

**Pull request merged**